dev_tools/utils/nxdl_utils.py

```diff
@@ -9,16 +9,32 @@
 from pathlib import Path
 from typing import List
 from typing import Optional
+from typing import Union
 
 import lxml.etree as ET
 from lxml.etree import ParseError as xmlER
 
 
-def decode_or_not(elem):
+def decode_or_not(elem: Union[bytes, str], decode: bool = True) -> str:
     """Decodes a byte array to string if necessary"""
+    __error_msg = f"Unsupported type {type(elem)}. Expected bytes, or str."
```

Comment:
The idea of the original function was that you can feed all types, and it would just return the same value, and only act on byte strings. Now, you have to include type selection everywhere. I don't see the clear advantage. There is anyways no proper type annotation and checking in these parts of the code.

Reply:
You are right, here we can just use byte decoding, so I'll remove the error. We wanted to try something similar in pynxtools, but only for string, bytes, and np arrays, so I got a bit confused. I'll change it back here.

Reply:
I changed it, so now the only changes are a bit of error checking, docs, and a new test.

Reply:
I saw the other PR as well. But there, similarly, a lot of type checking would need to be added everywhere in the code, which I don't clearly see the benefit of.

Reply:
I have also changed that one and removing type checking. I initially misinterpreted what you are using this function for, sorry.

Here, we have the same functionality as before, except list of bytes are also decoded. In the other PR in pynxtools, I am also adding support for decoding np array of strings (because NIAC does not want to have numpy in their requirements).